server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -258,16 +258,18 @@ protected void checkIfCanceled() {
   protected void announceExistence(HostAndPort clientAddress)
       throws KeeperException, InterruptedException {
 
-    String hostPort = ExternalCompactionUtil.getHostPortString(clientAddress);
-
     ZooReaderWriter zoo = getContext().getZooReaderWriter();
-    String compactorQueuePath =
-        getContext().getZooKeeperRoot() + Constants.ZCOMPACTORS + "/" + this.getResourceGroup();
-    String zPath = compactorQueuePath + "/" + hostPort;
 
+    final ServiceLockPath path =
+        getContext().getServerPaths().createCompactorPath(getResourceGroup(), clientAddress);
+    // The ServiceLockPath contains a resource group in the path which is not created
+    // at initialization time. If it does not exist, then create it.
+    final String compactorGroupPath =
+        path.toString().substring(0, path.toString().lastIndexOf("/" + path.getServer()));
+    LOG.debug("Creating compactor resource group path in zookeeper: {}", compactorGroupPath);
     try {
-      zoo.mkdirs(compactorQueuePath);
-      zoo.putPersistentData(zPath, new byte[] {}, NodeExistsPolicy.SKIP);
+      zoo.mkdirs(compactorGroupPath);
+      zoo.putPersistentData(path.toString(), new byte[] {}, NodeExistsPolicy.SKIP);
```

Comment:
No suggesting any changes for this PR.  Wondering how we could have common code for creating the ZK node for a RG. Seems like compactor, scan sever, and tablet server all need to do this.  Would be nice to avoid making assumptions about that path here and maybe some push this into ServiceLockPaths.